public/scripts/upload.js

```diff
@@ -228,7 +233,10 @@ class RootViewModel {
     this.selectedColorId = selectedColorId;
     this.maxMockupWaitSec = maxMockupWaitSec;
     this.fileList = fileListViewModel;
-    this.maxWorkers = navigator.hardwareConcurrency || 4;
+
+    this.maxWorkers = isUserAgentFirefox()
+      ? MAX_FIREFOX_WEB_WORKERS
+      : navigator.hardwareConcurrency;
```

Comment:
should also set to `MAX_FIREFOX_WEB_WORKERS` if `navigator.hardwareConcurrency` not support
https://caniuse.com/hardwareconcurrency